src/MassTransit/MetricsTagsFeature.cs

```diff
@@ -0,0 +1,11 @@
+﻿namespace MassTransit
+{
+    using System.Collections.Generic;
+
+    internal sealed class MetricsTagsFeature : IMetricsTagsFeature
+    {
+        ICollection<KeyValuePair<string, object>> IMetricsTagsFeature.Tags => TagsList;
+
+        public List<KeyValuePair<string, object>> TagsList { get; } = new List<KeyValuePair<string, object?>>();
```

Comment:
Why not just a TagList from the metrics library? Why the whole custom type? And dont' give access to the collection, just have the same methods Add, etc.

Reply:
A custom type can make things more flexible for future changes.
For example, this is how it started in aspnet v8.0.7.
https://github.com/dotnet/aspnetcore/blob/v8.0.7/src/Hosting/Hosting/src/Internal/HttpMetricsTagsFeature.cs

But in the v9 preview, there are additional things.
https://github.com/dotnet/aspnetcore/blob/v9.0.0-preview.6.24328.4/src/Hosting/Hosting/src/Internal/HttpMetricsTagsFeature.cs

I am fine with the direct Tags collection too. Please let me know what you think.